Drive/rover_drive_system.hpp

```diff
@@ -34,63 +44,130 @@ class RoverDriveSystem
 
   void Enable(bool enable = true)
   {
-    rover_is_operational_ = enable;
+    mission_control_data_.is_operational = true;
     left_wheel_.Enable(enable);
     right_wheel_.Enable(enable);
     back_wheel_.Enable(enable);
-    SetMode('S');
+    SetMode();
   }
 
-  /// Main function for handling all the rover drive system functionality. Might
-  /// not need to take in params since maybe GetMissionControlData() will be
-  /// responsible for grabbing commands and saving them within the
-  /// RoverDriveSystem class.
-  void Move(char mode,
-            units::angle::degree_t rotation_angle,
-            units::angular_velocity::revolutions_per_minute_t speed)
+  // Retrieves /drive Mission Control commands.
+  /// @return returns true if connection is established from mission control
+  bool GetMissionControlData()
   {
-    if (rover_is_operational_)
+    // GET /drive JSON value ?
+    if (true)  // change 'true' to something for localhost connection found
+    {
+      char response[] = {};
+      ParseMissionControlData(response);
+      Move(mission_control_data_.drive_mode,
+           mission_control_data_.rotation_angle, mission_control_data_.speed);
+      return true;
+    }
+    sjsu::LogError("Unable to reach mission control server");
+    return false;
+  }
+
+  /// Main function for handling all the rover drive system functionality.
+  /// @param mode Updates drive mode if new mode is different from current.
+  /// @param rotation_angle adjusts wheel position depending on mode.
+  /// @param speed adjusts the movement speed of the rover.
+  void Move(char mode, float rotation_angle, float speed)
+  {
+    if (mission_control_data_.is_operational)
     {
-      // checks if the mode is different than what is currently running
       if (mode != static_cast<char>(current_mode_))
       {
         SetMode(mode);
       }
-      HandleModeMovement(rotation_angle, speed);
+      HandleRoverMovement(rotation_angle, speed);
       UpdateMissionControlData();
     }
   };
 
-  // Homes all the wheels so the motors know their actual position. Returns true
-  // if successful.
-  bool Home()
+  /// Resets all the wheels so the motors know their actual position.
+  /// @return true if successfully resets wheels into start position
+  bool Reset()
   {
+    SetMode();
     return true;
   };
 
-  bool rover_is_operational_ = false;
-
  private:
-  ///  Sets the new mode for the rover. Will reduce rover movement speed to zero
-  ///  before changing wheel mode & tire alignment
-  void SetMode(char mode)
+  /// Sets all wheels to the speed provided. Wheel class handles max/min speeds
+  /// @param speed the new movement speed of the rover
+  void SetWheelSpeed(units::angular_velocity::revolutions_per_minute_t speed)
+  {
+    left_wheel_.SetSpeed(speed);
+    right_wheel_.SetSpeed(speed);
+    back_wheel_.SetSpeed(speed);
```

Comment:
Would probably be a good idea to throw an linear interploation (aka EMA aka exponential moving average) somewhere in here to smooth out changes in speed. The acceleration for the motors is stupid high and will cause issues with the robot. Just an FYI. A bit more complex for this change though. I'd leave a comment saying you'll fix this later. A issue to go along with it too will be nice as well.

Reply:
created issue and added TODO. Issue #153